### PR TITLE
Extensions and CodeSystems

### PR DESCRIPTION
**Pull request by omarisgreat**
Some resources have been added as examples for discussion with NHSE (e.g. mCODE extensions).
Some changes may also be reverted after discussion with UKCore e.g. extensions to RelatedPerson, but these have been added as a temporary measure to satisfy the requirements of the NHSE GU.

### CODE REVIEW

File: StructureDefinition/Extension-Genomics-Topology.StructureDefinition.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
+{
+  "resourceType" : "StructureDefinition",
+  "id" : "genomics-body-location-qualifier",
+  "url" : "https://fhir.nhs.uk/StructureDefinition/mcode-body-location-qualifier",
+  "version" : "2.0.0",
+  "name" : "BodyLocationQualifier",
+  "title" : "Body Location Qualifier Extension",
+  "status" : "active",
```
Comment (KazeemHamzat) [inline StructureDefinition/Extension-Genomics-Topology.StructureDefinition.json:8]: status: active or draft?

File: StructureDefinition/Extension-Genomics-Morphology.StructureDefinition.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
+{
+  "resourceType": "StructureDefinition",
+  "id": "genomics-histology-morphology-behavior",
+  "url": "https://fhir.nhs.uk/StructureDefinition/mcode-histology-morphology-behavior",
+  "version": "2.0.0",
+  "name": "HistologyMorphologyBehavior",
+  "title": "Histology-Morphology-Behavior Extension",
+  "status": "active",
```
Comment (KazeemHamzat) [inline StructureDefinition/Extension-Genomics-Morphology.StructureDefinition.json:8]: status: draft?

File: StructureDefinition/Extension-Genomics-Topology.StructureDefinition.json (hunk 1 above)
Comment (omarisgreat) [reply StructureDefinition/Extension-Genomics-Topology.StructureDefinition.json:8]: These extensions have been lifted from mCODE, which are active in that IG so I've marked them as active there (doesn't really make much difference to Genomics as everything needs to be tested anyway

**Review by KazeemHamzat — CHANGES_REQUESTED** [review]
Some of the Extensions and Value sets have their status as ACTIVE instead of DRAFT.

File: StructureDefinition/Extension-Genomics-Morphology.StructureDefinition.json (hunk 2 above)
Comment (omarisgreat) [reply StructureDefinition/Extension-Genomics-Morphology.StructureDefinition.json:8]: As above

**Comment by omarisgreat** [comment]
> Some of the Extensions and Value sets have their status as ACTIVE instead of DRAFT.

Hi Kazeem, as mentioned in the specific comments, these extensions/valuesets have been marked as active as they are active in the IG where they were taken from, which I wanted to reflect in this IG. Happy to change this if it is an issue though

File: Bundle/UKCore-Bundle-MichaelJonesRequest-Example_v3_message.json [hunk 3]
```diff
@@ -513,6 +513,16 @@
 			"resource": {
 				"resourceType": "ServiceRequest",
 				"id": "66c519dd-423f-4238-89c8-91c5be52f330",
+				"extension": [
```
Comment (omarisgreat) [inline Bundle/UKCore-Bundle-MichaelJonesRequest-Example_v3_message.json:516]: New Extension required in Genomics on ServiceRequest

File: CapabilityStatement/CapabilityStatement-genomics-requirements.json [hunk 4]
```diff
@@ -129,7 +129,7 @@
                 },
                 {
                     "type": "RelatedPerson",
-                    "profile": "https://fhir.hl7.org.uk/StructureDefinition/UKCore-RelatedPerson"
+                    "profile": "https://fhir.nhs.uk/StructureDefinition/NHSDigital-RelatedPerson-Genomics"
```
Comment (omarisgreat) [inline CapabilityStatement/CapabilityStatement-genomics-requirements.json:132]: Added new profile for RelatedPerson to support extensions, covering requirements from Genomics

File: CodeSystem/CodeSystem-Genomics-nhsnumber-unavailable-reason.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (omarisgreat) [inline CodeSystem/CodeSystem-Genomics-nhsnumber-unavailable-reason.json:1]: Example CodeSystem, to be updated after review from Genomics

File: CodeSystem/CodeSystem-Genomics-coverage.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (omarisgreat) [inline CodeSystem/CodeSystem-Genomics-coverage.json:1]: To categorise a ServiceRequest as NHS or private for reimbursement

File: StructureDefinition/Extension-Genomics-AdditionalContact.StructureDefinition.json [hunk 7]
```diff
@@ -0,0 +1,53 @@
+{
```
Comment (omarisgreat) [inline StructureDefinition/Extension-Genomics-AdditionalContact.StructureDefinition.json:1]: Contact for questions relating to a ServiceRequest where this differs from the requester

**Comment by KevinMayfield** [comment]
Many of the elements are constrained out. 

If this is a logical model then this is ok but I don't think this is good practice for an actual profile
We did this on STU3 and it caused issues

File: StructureDefinition/Extension-Genomics-BirthSex.StructureDefinition.json [hunk 8]
```diff
@@ -0,0 +1,78 @@
+{
```
Comment (omarisgreat) [inline StructureDefinition/Extension-Genomics-BirthSex.StructureDefinition.json:1]: Additional field required on RelatedPerson

File: StructureDefinition/Extension-Genomics-Deceased.StructureDefinition.json [hunk 9]
```diff
@@ -0,0 +1,45 @@
+{
```
Comment (omarisgreat) [inline StructureDefinition/Extension-Genomics-Deceased.StructureDefinition.json:1]: Additional field required on RelatedPerson

File: StructureDefinition/Extension-Genomics-EthnicCategory.StructureDefinition.json [hunk 10]
```diff
@@ -0,0 +1,78 @@
+{
```
Comment (omarisgreat) [inline StructureDefinition/Extension-Genomics-EthnicCategory.StructureDefinition.json:1]: Additional field required on RelatedPerson

File: StructureDefinition/Extension-Genomics-GPPractice.StructureDefinition.json [hunk 11]
```diff
@@ -0,0 +1,59 @@
+{
```
Comment (omarisgreat) [inline StructureDefinition/Extension-Genomics-GPPractice.StructureDefinition.json:1]: Additional field required on RelatedPerson

File: StructureDefinition/Extension-Genomics-Morphology.StructureDefinition.json (hunk 2 above)
Comment (omarisgreat) [reply StructureDefinition/Extension-Genomics-Morphology.StructureDefinition.json:8]: Extension from mCODE IG, covering requirements for definition of tumors in genomics

**Review by KevinMayfield — CHANGES_REQUESTED** [review]
The inclusion of Practitioner and Organization resources in the example Bundle look like they don't follow

https://simplifier.net/guide/NHSDigital/Home/Policy/Identifiers.guide.md?version=current

So when only a national identifier and name is present. Resource identifiers should be used.
This applies to Practitioner resources and probably Organization. 
We shouldn't be encouraging reference information to be distributed via messaging.

It makes it harder to developers to implement.

**Pull request merged**